develop/interoperability/intro-to-xcm.md

```diff
@@ -1,4 +1,110 @@
+
+* **BuyExecution**
+```rust
+--8<-- 'code/develop/parachain-devs/interoperability/XCM/XCM-buyexecution.rs'
+```
+    To execute XCM instructions, weight (some amount of resources) has to be bought. The amount of weight needed to execute an XCM depends on the number and type of instructions in the XCM. The `BuyExecution` instruction pays for the weight using the `fees`. The `fees` parameter describes the asset in the holding register that should be used for paying for the weight. The `weight_limit` parameter defines the maximum amount of fees that can be used for buying weight. There are special occasions where it is not necessary to buy weight. 
+
+* **DepositAsset**
+```rust
+--8<-- 'code/develop/parachain-devs/interoperability/XCM/XCM-depositasset.rs'
+```
+    The `DepositAsset` instruction is used to deposit funds from the holding register into the account of the beneficiary. You dont actually know how much is remaining in the holding register after the `BuyExecution` instruction, but that does not matter since you can specify a wildcard for the asset(s) which should be deposited.
+    
+    In this case, the wildcard is `All`, meaning that all assets in the holding register at that point in the execution should be deposited. The beneficiary in this case is the account of Bob in the current consensus system. 
+    
+    When the three instructions are combined, you withdraw `amount` native tokens from the account of Alice, pay for the execution of these instructions, and deposit the remaining tokens in the account of Bob.
+
+## Overview
+XCM enables different consensus systems to communicate with each other. Common cross-consensus use-cases include:
+
+* Sending tokens between blockchains
+* Locking assets on one blockchain in order to gain some benefit on a smart contract on another blockchain
+* Calling specific functions on another blockchain
+
+These are just a few basic examples; once you can communicate with other consensus systems, you can create applications that can leverage multiple blockchains' capabilities. The potential it provides is especially evident in an ecosystem of highly specialized blockchains like Polkadot.
+
+Decentralized distributed systems are very complex, so it's easy to make errors when building interactions between them. XCM is meant to be used by developers to package these interactions into their runtime logic before exposing that functionality to end users.
+
+XCM is constantly evolving; the format is expected to change over time. To keep up with the development of the format, or to propose changes, go to the [XCM format repository](https://github.com/polkadot-fellows/xcm-format).
```

Comment:
I'd add a small sentence that explains XCM versions (what's the current version used or something like that), since you are touching on the evolution of XCM